pkg/registry/extensions/deployment/storage/storage.go

```diff
@@ -225,6 +225,7 @@ func (r *ScaleREST) Update(ctx genericapirequest.Context, name string, objInfo r
 		return nil, false, err
 	}
 
+	// TODO: should we pass admission here?
```

Comment:
Yes, but you should do it by eliminating the use of the registry or changing the registry signature to accept the validation functions.

Reply:
Are the validation/update validation functions passed into this function going to be about Deployments or Scale types? I would naively expect the latter, in which case this code probably needs some drastic revision. (file an issue "scale (all?) subresources evade admission checking") rather than fix it in this PR.

Reply:
I would naively expect the former. As a webhook author, should I care about the way a user changes the replicas value? I would naively add an admission step for the full resource.

Reply:
> lavalamp 20 hours ago 
> Are the validation/update validation functions passed into this function going to be about Deployments or Scale types? I would naively expect the latter, in which case this code probably needs some drastic revision. (file an issue "scale (all?) subresources evade admission checking") rather than fix it in this PR.
>        
> sttts 2 hours ago 
> I would naively expect the former. As a webhook author, should I care about the way a user changes the replicas value? I would naively add an admission step for the full resource.

In our code today, the same admission chain is called for subresources as is called for normal resources.  This gives admission plugins the flexibility to decide if they care about subresource modifications (not all will), but if they do, it gives the full before/after context.

I think for now we should preserve the existing behavior (which I called out in a comment).  I'm open to discussing a change in behavior, but I'd like to do it outside the context of this pull.  @lavalamp I think you agree based on "(file an issue "scale (all?) subresources evade admission checking") rather than fix it in this PR."

Reply:
Comment is outdated. Will remove the TODO.